packages/doc/src/Cache.ts

```diff
@@ -0,0 +1,52 @@
+import fsCache from "file-system-cache";
+import { hashString } from "./utils";
+
+type HashPair = { sourceHash: string; commentStoreHash: string | undefined };
+
+/**
+ * Manages the cache for joist-doc
+ *
+ * joist-doc cares about two things per integration:
+ * - has the source changed
+ * - has the commentStore changed
+ *
+ * Cache holds a manifest document with these values as an object like so:
+ * Record<SourcePath, [SourceHash, CommentStoreHash]>
+ *
+ * If these, match, the previous version of the output can be restored,
+ * which is loaded separately under hash(SourcePath)
+ */
+export class Cache {
+  private fsCache = fsCache({
+    ns: "joist-doc",
+  });
+
+  private manifest: Record<string, [string, string | undefined]>;
+
+  constructor() {
+    try {
+      this.manifest = JSON.parse(this.fsCache.getSync("manifest"));
+    } catch {
+      this.manifest = {};
+    }
+  }
+
+  save() {
+    this.fsCache.setSync("manifest", JSON.stringify(this.manifest));
+  }
+
+  async set(filePath: string, { sourceHash, commentStoreHash }: HashPair, generated: string) {
+    this.manifest[filePath] = [sourceHash, commentStoreHash];
+    await this.fsCache.set(hashString(filePath), generated);
```

Comment:
Just curious, but why are we hashing the filePath before using it as a key? Is it to avoid special/weird characters that `fsCache` would not like as its key values?

Reply:
Correct! I'll try it without though because this was just an assumption; looks like file-system-cache is doing its own encoding of the filenames so this might not be required.

Reply:
Works without, removed!

Reply:
Sweet!